woodwork/deserialize.py

```diff
@@ -215,3 +182,98 @@ def _check_schema_version(saved_version_str):
             OutdatedSchemaWarning().get_warning_message(saved_version_str),
             OutdatedSchemaWarning,
         )
+
+
+def _typing_info_to_init_params(typing_info):
+    """Take table typing info created from the `typing_info_to_dict` function and extract
+    parameters that can be used to initialize Woodwork."""
+    logical_types = {}
+    semantic_tags = {}
+    column_descriptions = {}
+    column_origins = {}
+    column_metadata = {}
+    use_standard_tags = {}
+    for col in typing_info["column_typing_info"]:
+        col_name = col["name"]
+
+        ltype_metadata = col["logical_type"]
+        ltype = ww.type_system.str_to_logical_type(
+            ltype_metadata["type"], params=ltype_metadata["parameters"]
+        )
+
+        tags = col["semantic_tags"]
+
+        if "index" in tags:
+            tags.remove("index")
+        elif "time_index" in tags:
+            tags.remove("time_index")
+
+        logical_types[col_name] = ltype
+        semantic_tags[col_name] = tags
+        column_descriptions[col_name] = col["description"]
+        column_origins[col_name] = col["origin"]
+        column_metadata[col_name] = col["metadata"]
+        use_standard_tags[col_name] = col["use_standard_tags"]
+
+    return {
+        "name": typing_info.get("name"),
+        "index": typing_info.get("index"),
+        "time_index": typing_info.get("time_index"),
+        "logical_types": logical_types,
+        "semantic_tags": semantic_tags,
+        "column_descriptions": column_descriptions,
+        "column_origins": column_origins,
+        "table_metadata": typing_info.get("table_metadata"),
+        "column_metadata": column_metadata,
+        "use_standard_tags": use_standard_tags,
+    }
+
+
+def read_parquet(path, filename, lib="pandas", profile_name=None):
+    """
+    Read data from the parquet file at the location specified by ``path``
+    with the filename specified by ``filename``. Will initialize Woodwork using
+    the typing information stored in the parquet file metadata. If typing info
```

Comment:
Would it be worth indicating that it looks for. the woodwork typing info under the `"woodwork_metadata"` string? If someone has serialized and deserialized through woodwork, there's no real reason to have to know this, but it could be useful if there's some further manipulation a user wanted to do.  (Though maybe we specifically would want to avoid that, in which case, we shouldn't overly advertise location where the typing info is stored within the metadata

Reply:
I'm a bit on the fence here, but generally I am leaning toward not advertising this too much, as I don't think the intent would be for the users to do anything with the Woodwork metadata that gets stored. One reason I could see to list this would be to help advanced users avoid accidentally corrupting the WW data if they modify the parquet file metadata.

I'm thinking it might be best to update the key to include an underscore to better indicate that it isn't supposed to be modified by the users. Changes there, if not done properly, could impact the ability to deserialize the file, and isn't something I would expect casual users to do. 

Maybe we can revisit in the future if we find a situation where users need to know about the details, but for now I think I'd leave this out of the documentation. You ok with this?

Reply:
I like adding an underscore to the key a lot! And maybe a more genera note that users should be careful to not modify the parquet file metadata, as it may impact deserialization.